src/org/infinity/search/ScriptReferenceSearcher.java

```diff
@@ -44,17 +52,18 @@ protected void search(ResourceEntry entry, Resource resource)
 
   private void searchStruct(ResourceEntry entry, AbstractStruct struct)
   {
+    final String name = targetEntry.getResourceName();
     for (int i = 0; i < struct.getFieldCount(); i++) {
-      StructEntry o = struct.getField(i);
+      final StructEntry o = struct.getField(i);
       if (o instanceof ResourceRef &&
-          ((ResourceRef)o).getResourceName().equalsIgnoreCase(targetEntry.toString())) {
-        ResourceRef ref = (ResourceRef)o;
+          ((ResourceRef)o).getResourceName().equalsIgnoreCase(name)) {
         if (struct instanceof CreResource) {
-          addHit(entry, entry.getSearchString(), ref);
+          addHit(entry, entry.getSearchString(), o);
         } else if (struct instanceof Actor) {
-          addHit(entry, struct.getField(20).toString(), ref);
+          final IsTextual actorName = (IsTextual)struct.getAttribute(Actor.ARE_ACTOR_NAME);
```

Comment:
I think, in this place there was an error since field 20 is a reference to a script